src/java/htsjdk/samtools/BAMFileSpan.java

```diff
@@ -0,0 +1,268 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package htsjdk.samtools;
+
+import htsjdk.samtools.util.StringUtil;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * An ordered list of chunks, capable of representing a set of discontiguous
+ * regions in the BAM file.  FileSpans are mutable within the package, but perceived
+ * as immutable outside the package.
+ *
+ * Some operations on FileSpans assume that the spans are sorted.  In these cases,
+ * sort order will be validated.
+ *
+ * @author mhanna
+ * @version 0.1
+ */
+public class BAMFileSpan implements SAMFileSpan, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * The constituent chunks of this list.
+     */
+    private final List<Chunk> chunks;
+
+    /**
+     * Create a new empty list of chunks.
+     */
+    public BAMFileSpan() {
+        this.chunks = new ArrayList<Chunk>();
+    }
+
+    /**
+     * Convenience constructor to construct a BAM file span from
+     * a single chunk.
+     * @param chunk Chunk to use as the sole region in this span.
+     */
+    public BAMFileSpan(final Chunk chunk) {
+        this.chunks = new ArrayList<Chunk>();
+        chunks.add(chunk);
+    }
+
+    /**
+     * Create a new chunk list from the given list of chunks.
+     * @param chunks Constituent chunks.
+     */
+    public BAMFileSpan(final List<Chunk> chunks) {
+        this.chunks = new ArrayList<Chunk>(chunks);
+    }
+
+    /**
+     * Does this chunk list map to any position within the BAM file?
+     * @return True iff the ChunkList points to any data within the BAM.
+     */
+    public boolean isEmpty() {
+        return chunks.isEmpty();
+    }
+
+    /**
+     * Deep clone the given chunk list.
+     * @return A copy of the chunk list.
+     */
+    public BAMFileSpan clone() {
+        final BAMFileSpan clone = new BAMFileSpan();
+        for(final Chunk chunk: chunks)
+            clone.chunks.add(chunk.clone());
+        return clone;
+    }
+
+    /**
+     * Remove all chunks in this file span before the given file span starts.
+     * If a chunk in the chunk list starts before and ends after the given
+     * chunk, the first portion of the chunk will be deleted.
```

Comment:
The description for this public method makes it sounds as if it mutates the object, which (now that this class has been made public) would violate the contract that "FileSpans are mutable within the package, but perceived as immutable outside the package." Could you re-word this method description to make it clear that it returns a new file span without mutating the existing one?